runtime/shared_common/shrinit.cpp

```diff
@@ -3225,7 +3272,8 @@ j9shr_init(J9JavaVM *vm, UDATA loadFlags, UDATA* nonfatal)
 		parseResult==RESULT_DO_PRINTSTATS_EQUALS) {
 		doPrintStats = true;
 		/* Do not try to kill a cache if we just want to get stats on it */
-		runtimeFlags |= J9SHR_RUNTIMEFLAG_ENABLE_STATS;
+		/* set J9SHR_RUNTIMEFLAG_ENABLE_READONLY. If not set, vmCntr will be increased in the cache header */
```

Comment:
Is this an existing bug?

Reply:
I don't think it is a bug. When running printStats, previously we start up the cache in read write mode and increase the vmCntr, but no other data is added to the cache. I think this behavior was fine in the one cache world.

But in the case of multi-layer cache, we do not want any modification to the lower layer cache. printStats shouldn't modify the cache. So I add J9SHR_RUNTIMEFLAG_ENABLE_READONLY here. The only difference is vmCntr won't be increased anymore by printStats.